physics/src/IceGrowth.cpp

```diff
@@ -132,18 +137,22 @@ ConfigMap IceGrowth::getConfiguration() const
 
 void IceGrowth::update(const TimestepTime& tsTime)
 {
-
     // Copy the ice data from the prognostic fields to the modifiable fields.
     cice = cice0;
     overElements(std::bind(&IceGrowth::initializeThicknesses, this, std::placeholders::_1,
                      std::placeholders::_2),
         tsTime);
 
-    iVertical->update(tsTime);
-    // new ice formation
-    overElements(
-        std::bind(&IceGrowth::updateWrapper, this, std::placeholders::_1, std::placeholders::_2),
-        tsTime);
+    // reset the new ice volume array
+    newice = 0;
```

Comment:
I don't understand why ``newice`` is set to zero here. Isn't it enough to do it in line 195? And if we still need to initialise it, shouldn't that be done in ``initializeThicknesses``?

Reply:
The tricky thing was this at line 195 
```
       newice[0] = 0;
```
This has now been corrected with the commit that fixes it in other branches.

Setting `newice`to zero there ensures that it is zero even if `doThermo` is false. In that case, the code at line 195 is never approaches, as the `lateralIceSpread` function is never called.

In any case, the zeroing has been moved to `initializeIceThicknesses` it does make sense there.

Reply:
Ah, and I was looking at that line and didn't see the bug! :D Nice fix!